ApiDocs.Publishing/CSDL/CsdlWriter.cs

```diff
@@ -0,0 +1,497 @@
+            {
+                await writer.WriteAsync(xmlData);
+                await writer.FlushAsync();
+                writer.Close();
+            }
+        }
+
+        private EntityFramework CreateEntityFrameworkFromDocs()
+        {
+            var edmx = new EntityFramework();
+            
+            // Add resources
+            foreach (var resource in Documents.Resources)
+            {
+                var targetSchema = FindOrCreateSchemaForNamespace(resource.Name.NamespaceOnly(), edmx);
+                if (targetSchema != null)
+                {
+                    AddResourceToSchema(targetSchema, resource, edmx);
+                }
+            }
+
+            // Figure out the EntityCollection
+            BuildEntityCollection(edmx);
+
+            // Add actions to the collection
+            this.ProcessRestRequestPaths(edmx);
+
+            return edmx;
+        }
+
+        /// <summary>
+        /// Scan the MethodDefintions in the documentation and create actions and functions in the 
+        /// EntityFramework for matching call patterns.
+        /// </summary>
+        /// <param name="edmx"></param>
+        private void ProcessRestRequestPaths(EntityFramework edmx)
+        {
+            Dictionary<string, MethodCollection> uniqueRequestPaths = GetUniqueRequestPaths();
+
+            foreach (var path in uniqueRequestPaths.Keys)
+            {
+                var methodCollection = uniqueRequestPaths[path];
+
+                ODataTargetInfo requestTarget = null;
+                try
+                {
+                    requestTarget = ParseRequestTargetType(path, methodCollection, edmx);
+                    if (requestTarget.Classification == ODataTargetClassification.Unknown &&
+                        !string.IsNullOrEmpty(requestTarget.Name) &&
+                        requestTarget.QualifiedType != null)
+                    {
+                        CreateNewActionOrFunction(edmx, methodCollection, requestTarget);
+                    }
+                    else if (requestTarget.Classification == ODataTargetClassification.EntityType)
+                    {
+                        // We've learned more about this entity type, let's add that information to the state
+                        AppendToEntityType(edmx, requestTarget, methodCollection);
+                    }
+                    else if (requestTarget.Classification == ODataTargetClassification.NavigationProperty)
+                    {
+                        // TODO: Record somewhere the operations that are available on this NavigationProperty
+                        AppendToNavigationProperty(edmx, requestTarget, methodCollection);
+                    }
+                    else
+                    {
+                        // TODO: Are there interesting things to learn here?
+                        Console.WriteLine("Found type {0}: {1}", requestTarget.Classification, path);
```

Comment:
If we hit this condition, does that mean we didn't process something correctly and should bubble out an error?